### PR TITLE
Make checksumPeers more general

### PR DESCRIPTION
**Pull request by horizonzy**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>.

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
lgtm

File: jraft-core/src/main/java/com/alipay/sofa/jraft/entity/Checksum.java [hunk 1]
```diff
@@ -40,4 +42,21 @@
...
+     * Returns the checksum value of act on factors.
+     *
+     * @param factors checksum collection
+     * @param v origin checksum
+     * @return checksum value
+     */
+    default long checksum(final Collection<? extends Checksum> factors, long v) {
+        if (factors != null && !factors.isEmpty()) {
+            for (final Checksum factor : factors) {
+                v = checksum(factor.checksum(), v);
```
Comment (killme2008) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/entity/Checksum.java:56]: 这个 MR 改变了 checksum 的计算，原来是  `c = checksum(c, peer.checksum())` ， v 在前，现在是放到后面，可能有兼容性问题。

**Comment by killme2008** [comment]
@horizonzy @fengjiachun 建议再提个 PR 修正下这个 checkum 计算顺序，我不确定是否有兼容问题，最好不要改变。

**Comment by horizonzy** [comment]
> @horizonzy @fengjiachun 建议再提个 PR 修正下这个 checkum 计算顺序，我不确定是否有兼容问题，最好不要改变。

嗯，不改变原有逻辑。之前重写大意了。

**Pull request merged**